src/v/coproc/wasm_event.cc

```diff
@@ -59,6 +72,22 @@ get_value_for_event_header(const model::record& r, event_header header) {
       [&match](const model::record_header& rh) { return rh.key() == match; });
 }
 
+std::variant<wasm::errc, event_type> get_coproc_type(const model::record& r) {
```

Comment:
all that std::variant business looks a bit strange to me
if this code returns an event type or alternatively an error, it should probably just return an event type and throw on error, or it can return result<> or boost::outcome<>

wasm::errc might be integrated with std::error_code so it could work with result<>

Reply:
I tried do not change old logic in this place
Maybe @graphcareful  can help with this logic (but he is OOO on this week)...

Reply:
For me it is the same things (throw exception, return error_code)

Reply:
done

Reply:
I think its ok to change old logic as long as its done in a new commit